routers/web/repo/download.go

```diff
@@ -79,34 +82,57 @@ func ServeBlobOrLFS(ctx *context.Context, blob *git.Blob) error {
 	return common.ServeBlob(ctx, blob)
 }
 
+func getBlobForEntry(ctx *context.Context) *git.Blob {
+	entry, err := ctx.Repo.Commit.GetTreeEntryByPath(ctx.Repo.TreePath)
+	if err != nil {
+		ctx.ServerError("GetBlobByPath", err)
+		return nil
+	}
+
+	if entry.IsDir() || entry.IsSubModule() {
```

Comment:
Merge these 2 conditionals to line 87?

Reply:
One is a ServerError but this is a NotFound.